packages/cw-orch-core/src/contract/interface_traits.rs

```diff
@@ -206,7 +213,10 @@ pub trait CallAs<Chain: CwEnv>: CwOrchExecute<Chain> + ContractInstance<Chain> +
     }
 }
 
-impl<T: CwOrchExecute<Chain> + ContractInstance<Chain> + Clone, Chain: CwEnv> CallAs<Chain> for T {}
+impl<T: CwOrchExecute<Chain> + ContractInstance<Chain> + Clone, Chain: TxHandler> CallAs<Chain>
+    for T
+{
+}
 
 /// Helper methods for conditional uploading of a contract.
 pub trait ConditionalUpload<Chain: CwEnv>: CwOrchUpload<Chain> {
```

Comment:
Can we loosen this bound as well? Could you search for all the uses of `CwEnv` in this library and replace them with their minimal requirements?

Reply:
That's what I did yes ! And we can't loosen that bound because you need both QueryHandler and TxHandler to be able to use this trait !

Reply:
emoved in the proc macros as well :)